apps/mobile/src/services/notifications/NotificationService.ts

```diff
@@ -0,0 +1,273 @@
+import notifee, {
+  AuthorizationStatus,
+  Event as NotifeeEvent,
+  EventType,
+  EventDetail,
+  AndroidChannel,
+} from '@notifee/react-native'
+import { Linking, Platform, Alert as NativeAlert } from 'react-native'
+import { store } from '@/src/store'
+import { updatePromptAttempts, updateLastTimePromptAttempted } from '@/src/store/notificationsSlice'
+import { toggleAppNotifications, toggleDeviceNotifications } from '@/src/store/notificationsSlice'
+
+import { HandleNotificationCallback, LAUNCH_ACTIVITY, PressActionId } from '@/src/store/constants'
+
+import { ChannelId, notificationChannels, withTimeout } from '@/src/utils/notifications'
+import Logger from '@/src/utils/logger'
+
+import { FirebaseMessagingTypes } from '@react-native-firebase/messaging'
+
+interface AlertButton {
+  text: string
+  onPress: () => void | Promise<void>
+}
+
+class NotificationsService {
+  async getBlockedNotifications(): Promise<Map<ChannelId, boolean>> {
+    try {
+      const settings = await notifee.getNotificationSettings()
+      const channels = await notifee.getChannels()
+
+      switch (settings.authorizationStatus) {
+        case AuthorizationStatus.NOT_DETERMINED:
+        case AuthorizationStatus.DENIED:
+          // @ts-ignore
+          return notificationChannels.reduce((map, next) => {
+            map.set(next.id as ChannelId, true)
+            return map
+          }, new Map<ChannelId, boolean>())
+      }
+
+      return channels.reduce((map, next) => {
+        if (next.blocked) {
+          map.set(next.id as ChannelId, true)
+        }
+        return map
+      }, new Map<ChannelId, boolean>())
+    } catch (e) {
+      Logger.error('Error checking if a user has push notifications permission', e)
+      return new Map<ChannelId, boolean>()
+    }
+  }
+
+  async getAllPermissions(shouldOpenSettings = true) {
+    const promises: Promise<string>[] = notificationChannels.map((channel: AndroidChannel) =>
+      withTimeout(this.createChannel(channel), 5000),
+    )
+    // 1 - Creates android's notifications channel
+    await Promise.allSettled(promises)
+    // 2 - Verifies granted permission from device
+    let permission = await withTimeout(this.checkCurrentPermissions(), 5000)
+    // 3 - Verifies blocked notifications
+    const blockedNotifications = await withTimeout(this.getBlockedNotifications(), 5000)
+    /**
+     * 4 - If permission has not being granted already or blocked notifications are found, open device's settings
+     * so that user can enable DEVICE notifications
+     **/
+    if ((permission !== 'authorized' || blockedNotifications.size !== 0) && shouldOpenSettings) {
+      await this.requestPushNotificationsPermission()
+      permission = await withTimeout(this.checkCurrentPermissions(), 5000)
+    }
+    return { permission, blockedNotifications }
```

Comment:
Shall we add a try/catch block here since the requests can take more than the specified timeout and fail?